src/pseudopeople/entity_types.py

```diff
@@ -1,20 +1,22 @@
 from abc import ABC, abstractmethod
 from dataclasses import dataclass
-from typing import Any, Callable, Dict, List, Optional, Tuple
+from typing import TYPE_CHECKING, Any, Callable, Dict, List, Optional
 
 import pandas as pd
 from loguru import logger
 from vivarium import ConfigTree
-from vivarium.framework.randomness import RandomnessStream
 
 from pseudopeople.configuration import Keys
 from pseudopeople.utilities import get_index_to_noise
 
+if TYPE_CHECKING:
+    from pseudopeople.dataset import DatasetData
+
 
 @dataclass
```

Comment:
The changes in this file are all either updating arguments in calls or cleaning up some of the code that now lives in the new class.